internal/mithril-persistence/src/sqlite/connection_extensions.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::Context;
+use sqlite::{ReadableWithIndex, Value};
+
+use mithril_common::StdResult;
+
+use crate::sqlite::SqliteConnection;
+
+/// Extension trait for the [SqliteConnection] type.
+pub trait ConnectionExtensions {
+    /// Execute the given sql query and return the value of the first cell read.
+    fn query_single_cell<Q: Into<String>, T: ReadableWithIndex>(
```

Comment:
Since it's a low level api that may be used in different kind of scenario I would like to have some flexibility with the input parameters, hence the generic.
This said after re-challenging it I don't need a owned string here but only a ref to a str, so I'm thinking of changing the generics to `AsRef<str>` this avoid an allocation and simplify a bit the implementation:
```diff
 impl ConnectionExtensions for SqliteConnection {
-    fn query_single_cell<Q: Into<String>, T: ReadableWithIndex>(
+    fn query_single_cell<Q: AsRef<str>, T: ReadableWithIndex>(
         &self,
         sql: Q,
         params: &[Value],
     ) -> StdResult<T> {
-        let sql = sql.into();
         let mut statement = self.prepare(&sql).with_context(|| {
             format!(
                 "Prepare query error: SQL=`{}`",
-                &sql.replace('\n', " ").trim()
+                sql.as_ref().replace('\n', " ").trim()
             )
         })?;
         statement.bind(params)?;
```